<launch>
    <!-- General Control System Parameters -->
    <arg name="loop_rate" default="10.0"/>
    <arg name="min_linear_val" default="-5.0"/>
    <arg name="max_linear_val" default="5.0"/>
    <arg name="min_angular_val" default="-180.0"/>
    <arg name="max_angular_val" default="180.0"/>

    <!-- Velocity X Control System (Forward/Backwards) -->
    <arg name="Kp_vel_x" default="0.1"/>
    <arg name="Ki_vel_x" default="0.01"/>

    <!-- Velocity Y Control System (Left/Right) -->
    <arg name="Kp_vel_y" default="0.1"/>
    <arg name="Ki_vel_y" default="0.01"/>
   
    <!-- Position Z Control System (Depth/Up/Down) -->
    <arg name="Kp_pos_z" default="0.1"/>
    <arg name="Ki_pos_z" default="0.01"/>
    <arg name="Kp_vel_z" default="0.1"/>
    <arg name="Ki_vel_z" default="0.01"/>

    <!-- Position Pitch Control System -->
    <arg name="Kp_pos_p" default="0.1"/>
    <arg name="Ki_pos_p" default="0.01"/>
    <arg name="Kp_vel_p" default="0.1"/>
    <arg name="Ki_vel_p" default="0.01"/>

    <!-- Position Roll Control System -->
    <arg name="Kp_ros_r" default="0.1"/>
    <arg name="Ki_ros_r" default="0.01"/>
    <arg name="Kp_vel_r" default="0.1"/>
    <arg name="Ki_vel_r" default="0.01"/>
    
    <!-- Velocity Yaw Control System -->
    <arg name="Kp_vel_yw" default="0.1"/>
    <arg name="Ki_vel_yw" default="0.01"/>
    
    <node name ="rov" pkg="navigation" type="rov" required="true">
        <param name="loop_rate" value="$(arg loop_rate)"/>
        <param name="min_linear_val" value="$(arg min_linear_val)"/>
        <param name="max_linear_val" value="$(arg max_linear_val)"/>
        <param name="min_angular_val" value="$(arg min_angular_val)"/>
        <param name="max_angular_val" value="$(arg max_angular_val)"/>
        <param name="Kp_vel_x" value="$(arg Kp_vel_x)"/>
        <param name="Ki_vel_x" value="$(arg Ki_vel_x)"/>
        <param name="Kp_vel_y" value="$(arg Kp_vel_y)"/>
        <param name="Ki_vel_y" value="$(arg Ki_vel_y)"/>
        <param name="Kp_pos_z" value="$(arg Kp_pos_z)"/>
        <param name="Ki_pos_z" value="$(arg Ki_pos_z)"/>
        <param name="Kp_vel_z" value="$(arg Kp_vel_z)"/>
        <param name="Ki_vel_z" value="$(arg Ki_vel_z)"/>
        <param name="Kp_pos_p" value="$(arg Kp_pos_p)"/>
        <param name="Ki_pos_p" value="$(arg Ki_pos_p)"/>
        <param name="Kp_vel_p" value="$(arg Kp_vel_p)"/>
        <param name="Ki_vel_p" value="$(arg Ki_vel_p)"/>
        <param name="Kp_ros_r" value="$(arg Kp_ros_r)"/>
        <param name="Ki_ros_r" value="$(arg Ki_ros_r)"/>
        <param name="Kp_vel_r" value="$(arg Kp_vel_r)"/>
        <param name="Ki_vel_r" value="$(arg Ki_vel_r)"/>
        <param name="Kp_vel_yw" value="$(arg Kp_vel_yw)"/>
        <param name="Ki_vel_yw" value="$(arg Ki_vel_yw)"/>
    </node>

</launch>
